swift-sdk/DeviceInfo.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  DeviceInfo.swift
+//  swift-sdk
+//
+//  Created by Tapash Majumder on 7/26/18.
+//  Copyright © 2018 Iterable. All rights reserved.
+//
+
+import Foundation
+
+struct DeviceInfo : Codable {
+    let mobileDeviceType = MobileDeviceType.iOS
+    let deviceFp: DeviceFp
+    
+    enum MobileDeviceType : String, Codable {
+        case iOS
+        case Android
+    }
+    
+    struct DeviceFp : Codable {
+        let iosDeviceType: String
+        let screenWidth: String
+        let screenHeight: String
+        let screenScale: String
+        let version: String // systemVersion iOS
+        let timezoneOffsetMinutes: String
+        let language: String // current locale
+    }
+    
+    static func createDeviceInfo() -> DeviceInfo {
+        return DeviceInfo(deviceFp: createDeviceFp())
+    }
+    
+    private static func createDeviceFp() -> DeviceFp {
+        let screen = UIScreen.main
+        let device = UIDevice.current
+        // secondsFromGMT = time - GMT, timeZoneOffset = GMT - time
+        let secondsFromGMT = TimeZone.current.secondsFromGMT()
+        let timezoneOffsetMinutes = (-1.0 * Float(secondsFromGMT) / 60.0)
+        return DeviceFp(iosDeviceType: getModelName(),
+                          screenWidth: String(Float(screen.bounds.width)),
+                          screenHeight: String(Float(screen.bounds.height)),
+                          screenScale: String(Float(screen.scale)),
+                          version: device.systemVersion,
+                          timezoneOffsetMinutes: String(timezoneOffsetMinutes),
+                          language: Locale.current.identifier)
+    }
+
+    private static func getModelName() -> String {
+        var systemInfo = utsname()
+        uname(&systemInfo)
+        let machineMirror = Mirror(reflecting: systemInfo.machine)
+        let identifier = machineMirror.children.reduce("") { identifier, element in
+            guard let value = element.value as? Int8, value != 0 else { return identifier }
+            return identifier + String(UnicodeScalar(UInt8(value)))
+        }
+        
+        switch identifier {
```

Comment:
For the devices that we can map, we return the known names such as 'iPhone 7 Plus' as  you have noticed. For the ones that are not known we will still get the identifier "iPhone10,3" or "iPhone11,2"... something like that. 

Right now how the matching logic works is that... .the device sends all information that it has and Javascript sends all information that it has. Then the matching engine tries to match the two. The model names are not important for the matching engine at the moment. It will look at the screen size, os version, screen width, height, ipAddress time etc. The model information is not used right but but it is there and may be used in future matching strategies. 

I can add iPhone X details in here. The caveat is that we won't have iPhone XS when it comes out.

Reply:
I am beginning to think passing this information does not provide a lot of value at the moment so perhaps I can remove this. We can add this back later if needed. That will lead to less confusion. We can just use `UserInterfaceIdiom` which is provided by the system.